app/upgrades/v18/upgrades.go

```diff
@@ -0,0 +1,216 @@
+package v17
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	recordskeeper "github.com/Stride-Labs/stride/v17/x/records/keeper"
+	recordtypes "github.com/Stride-Labs/stride/v17/x/records/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v17/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v17/x/stakeibc/types"
+)
+
+func NewIntFromString(val string) sdkmath.Int {
+	out_int, ok := sdkmath.NewIntFromString(val)
+	if !ok {
+		panic(fmt.Sprintf("invalid string %s", val))
+	}
+	return out_int
+}
+
+var (
+	UpgradeName = "v18"
+
+	// Redemption rate bounds updated to give ~3 months of slack on outer bounds
+	RedemptionRateOuterMinAdjustment = sdk.MustNewDecFromStr("0.05")
+	RedemptionRateOuterMaxAdjustment = sdk.MustNewDecFromStr("0.10")
+
+	// Osmosis will have a slighly larger buffer with the redemption rate
+	// since their yield is less predictable
+	OsmosisChainId              = "osmosis-1"
+	OsmosisRedemptionRateBuffer = sdk.MustNewDecFromStr("0.02")
+
+	// Get Initial Redemption Rates for Unbonding Records Migration
+	InitialRedemptionRates = map[string]sdk.Dec{
+		"comdex-1":     sdk.MustNewDecFromStr("1.199107579864670291"),
```

Comment:
did you determine these by doing division on the records or by doing a historical query on the host zone?

Reply:
or is this supposed to be as refreshed as possible and we'll replace it with the RR's at the time the prop goes up?

Reply:
The latter! We'll refresh these with the live RRs at the time of the prop (we have a script that will refresh all of the constants at the top of this file)